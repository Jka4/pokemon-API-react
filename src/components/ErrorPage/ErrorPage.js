import React from "react";
import './styles/Style.scss'

const ErrorPage = () => {

	return (
		<div className='errorPage' >
			<div className="errCode">404</div>
			<div className="image">
				<img src="https://github.com/Jka4/pokemon-API-react/blob/master/public/images/404.png?raw=true" alt="" />
			</div>
			<div className="description">Page Not Found</div>

		</div>
	);
}

export default ErrorPage;
