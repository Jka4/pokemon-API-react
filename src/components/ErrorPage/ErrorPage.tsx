import React from 'react';

import styled from 'styled-components/macro';

const ErrorPage: React.FC = () => {
  return (
    <ErrorPageStyled>
      <div className="errCode">404</div>
      <div className="image">
        <Image src={`${process.env.PUBLIC_URL}/images/404.png`} alt="404_pickachu" />
      </div>
      <div className="description">Page Not Found</div>
    </ErrorPageStyled>
  );
};

const ErrorPageStyled = styled.div`
  width: 100%;
  min-height: 700px;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;

  .errCode {
    font-size: 42px;
  }

  .description {
    font-size: 24px;
  }
`;

const Image = styled.img`
  filter: drop-shadow(13px 17px 45px #222);
`;

export default ErrorPage;
