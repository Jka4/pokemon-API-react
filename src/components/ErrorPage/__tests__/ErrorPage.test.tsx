import * as React from 'react';

import ErrorPage from '../ErrorPage';

import { mount, shallow } from 'enzyme';

describe('<ErrorPage/> ', () => {
  const container = mount(<ErrorPage />);

  it('renders correctly', () => {
    expect(container).toMatchSnapshot();
  });
});
