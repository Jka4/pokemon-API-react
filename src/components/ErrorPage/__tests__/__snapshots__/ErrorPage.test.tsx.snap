// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorPage/>  renders correctly 1`] = `
<ErrorPage>
  <styled.div>
    <div
      className="sc-bdfBwQ kfFOsf"
    >
      <styled.img
        alt="404_pikachu"
        src="undefined/images/404.png"
      >
        <img
          alt="404_pikachu"
          className="sc-hKgILt jFqASf"
          src="undefined/images/404.png"
        />
      </styled.img>
      <styled.div>
        <div
          className="sc-gsTCUz iPjwDd"
        >
          404
        </div>
      </styled.div>
      <styled.div>
        <div
          className="sc-dlfnbm KPaPX"
        >
          Page Not Found
        </div>
      </styled.div>
    </div>
  </styled.div>
</ErrorPage>
`;
